goodconf/__init__.py

```diff
@@ -65,26 +67,57 @@ def initial_for_field(name: str, field: ModelField) -> Any:
     return ""
 
 
+def file_config_settings_source(settings: BaseSettings) -> Dict[str, Any]:
+    """Find config file and get values"""
+    settings = cast(GoodConf, settings)
+    selected_config_file = None
+    if settings._config_file:
+        selected_config_file = settings._config_file
+    elif (
+        settings.__config__.file_env_var
+        and settings.__config__.file_env_var in os.environ
+    ):
+        selected_config_file = _find_file(os.environ[settings.__config__.file_env_var])
+    else:
+        for filename in settings.__config__.default_files or []:
+            selected_config_file = _find_file(filename, require=False)
+            if selected_config_file:
+                break
+    if selected_config_file:
+        values = _load_config(selected_config_file)
+        log.info("Loading config from %s", selected_config_file)
+        settings.__config__._config_file = selected_config_file
+    else:
+        values = {}
+        log.info("No config file specified. Loading with environment variables.")
+    return values
+
+
 class GoodConf(BaseSettings):
-    def __init__(self, load: bool = False):
+    def __init__(self, load: bool = False, **kwargs):
         """
         :param load: load config file on instantiation [default: False].
 
         A docstring defined on the class should be a plain-text description
         used as a header when generating a configuration file.
         """
-        if load:
-            self.load()
+        self._config_file = None
+        # Emulate Pydantic behavior, load immediately
+        if kwargs:
+            return super().__init__(**kwargs)
+        elif load:
+            return self.load()
+
+    _config_file: str | None = PrivateAttr()
```

Comment:
This requires Pydantic 1.7, if this is not okay for you then I can work around this.